ValheimPlus/GameClasses/Inventory.cs

```diff
@@ -80,4 +80,42 @@ public static void Prefix(string name, ref int w, ref int h)
 			}
 		}
 	}
+
+	/// <summary>
+	/// When merging another inventory, try to merge items with existing stacks.
+	/// </summary>
+	[HarmonyPatch(typeof(Inventory), "MoveAll")]
+	public static class Inventory_MoveAll_Patch
+	{
+		private static void Prefix(ref Inventory __instance, ref Inventory fromInventory)
+		{
+			if (Configuration.Current.Inventory.IsEnabled && Configuration.Current.Inventory.mergeWithExistingStacks)
+			{
+				List<ItemDrop.ItemData> list = new List<ItemDrop.ItemData>(fromInventory.GetAllItems());
+				foreach (ItemDrop.ItemData otherItem in list)
+				{
+					if (otherItem.m_shared.m_maxStackSize > 1)
+					{
+						foreach (ItemDrop.ItemData myItem in __instance.m_inventory)
+						{
+							if (myItem.m_shared.m_name == otherItem.m_shared.m_name && myItem.m_quality == otherItem.m_quality)
+							{
+								int itemsToMove = Math.Min(myItem.m_shared.m_maxStackSize - myItem.m_stack, otherItem.m_stack);
+								myItem.m_stack += itemsToMove;
```

Comment:
Should checks be done for weight?  I haven't played the game long enough but I imagine if we take a large stack of something heavy we'd have to go back into the inventory to drop that stack again to be able to run away from danger.

Reply:
Though I suppose the default might not check this weight, but then this makes it much more of a chance of picking up too much weight.

Reply:
The game currently doesn't check for weight. I've been thinking about this but I think we should add something smarter than just stop when you hit the weight limit. If we do add a check to stop at the weight capacity I would like it to prioritize important items such as weapons, armor and food. It would suck to only pick up your stack of ore and be left defenseless.